src/server/scripts/Commands/cs_lfg.cpp

```diff
@@ -64,48 +66,45 @@ class lfg_commandscript : public CommandScript
         return commandTable;
     }
 
-    static bool HandleLfgPlayerInfoCommand(ChatHandler* handler, char const* args)
+    static bool HandleLfgPlayerInfoCommand(ChatHandler* handler, Optional<PlayerIdentifier> player)
     {
-        Player* target = nullptr;
-        std::string playerName;
-        if (!handler->extractPlayerTarget((char*)args, &target, nullptr, &playerName))
+        if (!player)
+            player = PlayerIdentifier::FromTargetOrSelf(handler);
+        if (!player)
             return false;
 
-        GetPlayerInfo(handler, target);
-        return true;
+        if (Player* target = player->GetConnectedPlayer())
+        {
+            PrintPlayerInfo(handler, target);
+            return true;
+        }
+
+        return false;
     }
 
-    static bool HandleLfgGroupInfoCommand(ChatHandler* handler, char const* args)
+    static bool HandleLfgGroupInfoCommand(ChatHandler* handler, Optional<PlayerIdentifier> player)
     {
-        Player* playerTarget;
-        ObjectGuid guidTarget;
-        std::string nameTarget;
-
-        ObjectGuid parseGUID(HighGuid::Player, uint32(atoul(args)));
-
-        if (sCharacterCache->GetCharacterNameByGuid(parseGUID, nameTarget))
-        {
-            playerTarget = ObjectAccessor::FindPlayer(parseGUID);
-            guidTarget = parseGUID;
-        }
-        else if (!handler->extractPlayerTarget((char*)args, &playerTarget, &guidTarget, &nameTarget))
+        if (!player)
+            player = PlayerIdentifier::FromTargetOrSelf(handler);
+        if (!player)
             return false;
 
         Group* groupTarget = nullptr;
 
-        if (playerTarget)
-            groupTarget = playerTarget->GetGroup();
+        if (player->IsConnected())
```

Comment:
`if (Player* target = player->GetConnectedPlayer())`

Reply:
Does it make sense to introduce a local for that though I'm only interested in the group?

Reply:
Well, you use that same value in the next line, so...